xdsl/dialects/vector.py

```diff
@@ -292,6 +295,90 @@ def get(mask_operands: list[Operation | SSAValue]) -> CreatemaskOp:
         )
 
 
+@irdl_op_definition
+class ExtractElementOp(IRDLOperation):
+    name = "vector.extractelement"
+    vector = operand_def(VectorType)
+    position = opt_operand_def(IndexTypeConstr | SignlessIntegerConstraint)
+    result = result_def(Attribute)
+    traits = traits_def(Pure())
+
+    def verify_(self):
+        assert isa(self.vector.type, VectorType[Attribute])
+
+        if self.result.type != self.vector.type.element_type:
+            raise VerifyException(
+                "Expected result type to match element type of vector operand."
+            )
+
+        if self.vector.type.get_num_dims() == 0:
+            if self.position is not None:
+                raise VerifyException("Expected position to be empty with 0-D vector.")
+            return
+        if self.vector.type.get_num_dims() != 1:
+            raise VerifyException("Unexpected >1 vector rank.")
+        if self.position is None:
+            raise VerifyException("Expected position for 1-D vector.")
+
+    @staticmethod
+    def get(
+        vector: SSAValue | Operation,
+        position: SSAValue | Operation | None = None,
+    ):
+        vector = SSAValue.get(vector)
+        assert isa(vector.type, VectorType[Attribute])
+        result_type = vector.type.element_type
+
+        return ExtractElementOp.build(
+            operands=[vector, position],
+            result_types=[result_type],
+        )
```

Comment:
Could you please add a custom init here and below?

Reply:
I mean instead of a .get

Reply:
Done